exporters/otlp/otlphttp/driver.go

```diff
@@ -0,0 +1,269 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlphttp
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"path"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	colmetricspb "go.opentelemetry.io/otel/exporters/otlp/internal/opentelemetry-proto-gen/collector/metrics/v1"
+	coltracepb "go.opentelemetry.io/otel/exporters/otlp/internal/opentelemetry-proto-gen/collector/trace/v1"
+	"go.opentelemetry.io/otel/exporters/otlp/internal/transform"
+	metricsdk "go.opentelemetry.io/otel/sdk/export/metric"
+	tracesdk "go.opentelemetry.io/otel/sdk/export/trace"
+)
+
+const contentType = "application/x-protobuf"
+
+var defaultTransportClone *http.Transport
+
+func init() {
+	// Clone default transport early, possibly before it's
+	// used/modified.
+	realTransport := http.DefaultTransport.(*http.Transport)
+	defaultTransportClone = realTransport.Clone()
+}
+
+type driver struct {
+	client *http.Client
+	cfg    config
+
+	stopCh chan struct{}
+}
+
+var _ otlp.ProtocolDriver = (*driver)(nil)
+
+// NewDriver creates a new HTTP driver.
+func NewDriver(opts ...Option) otlp.ProtocolDriver {
+	cfg := config{
+		address:        fmt.Sprintf("%s:%d", otlp.DefaultCollectorHost, otlp.DefaultCollectorPort),
+		compression:    NoCompression,
+		tracesURLPath:  DefaultTracesPath,
+		metricsURLPath: DefaultMetricsPath,
+		maxAttempts:    DefaultMaxAttempts,
+		backoff:        DefaultBackoff,
+	}
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+	for pathPtr, defaultPath := range map[*string]string{
+		&cfg.tracesURLPath:  DefaultTracesPath,
+		&cfg.metricsURLPath: DefaultMetricsPath,
+	} {
+		tmp := strings.TrimSpace(*pathPtr)
+		if tmp == "" {
+			tmp = defaultPath
+		} else {
+			tmp = path.Clean(tmp)
+			if !path.IsAbs(tmp) {
+				tmp = fmt.Sprintf("/%s", tmp)
+			}
+		}
+		*pathPtr = tmp
+	}
+	if cfg.maxAttempts <= 0 {
+		cfg.maxAttempts = DefaultMaxAttempts
+	}
+	if cfg.maxAttempts > DefaultMaxAttempts {
+		cfg.maxAttempts = DefaultMaxAttempts
+	}
+	if cfg.backoff <= 0 {
+		cfg.backoff = DefaultBackoff
+	}
+	client := &http.Client{}
+	if cfg.tlsCfg != nil {
+		transport := defaultTransportClone.Clone()
+		transport.TLSClientConfig = cfg.tlsCfg
+		client.Transport = transport
```

Comment:
> Added lines #L98 - L100 were not covered by tests

Can we get a test for the TLS config handling?

Reply:
Done.